src/client/protocolgameparse.cpp

```diff
@@ -1451,37 +1463,59 @@ void ProtocolGame::parseDistanceMissile(const InputMessagePtr& msg)
 void ProtocolGame::parseItemClasses(const InputMessagePtr& msg)
 {
     const uint8_t classSize = msg->getU8();
-    for (uint8_t i = 0; i < classSize; i++) {
+    for (uint_fast8_t i = 0; i < classSize; i++) {
         msg->getU8(); // class id
 
         // tiers
         const uint8_t tiersSize = msg->getU8();
-        for (uint8_t j = 0; j < tiersSize; j++) {
+        for (uint_fast8_t j = 0; j < tiersSize; j++) {
```

Comment:
If you want it to be fast, just use `auto` or `decltype(tiersSize)` to avoid conversions

Reply:
thank you @ranisalt ranisal